File: clojure/hamming/src/hamming.clj
(ns hamming)

(defn hamming-occurs? [strand1 strand2 i]
  (let [c1 (.charAt strand1 i)
        c2 (.charAt strand2 i)
        occurs (not (= c1 c2))
        ]
    occurs))

(defn indexes [n] (range n))

(defn distance [strand1 strand2]
  (if (= strand1 strand2)
    0
    (if (= (count strand1) (count strand2))
      (count (filter identity (map #(hamming-occurs? strand1 strand2 %) (indexes (count strand1))))))))

File: solutions/ruby/hamming/1/hamming.rb
class Hamming
  def self.compute(strand1, strand2)
    raise ArgumentError if strand1.length != strand2.length

    (0..strand1.length).inject(0) do |count, n|
      count + (strand1[n] == strand2[n] ? 0 : 1)
    end
  end
end
